### ISSUE TITLE
Run on Desktop VM

### ISSUE THREAD

**Issue by dumazy** [flutter/flutter#79237]
## Use case

When targeting Linux, you also have to develop in a Linux environment. This means you have to switch development environments to target multiple platforms. If you want to support Linux and iOS, for example, you cannot just develop on macOS.

## Proposal

Would it be possible to support running on virtual machines or maybe even other connected machines? The developer could have a virtual machine for Ubuntu running while developing on macOS and targeting other platforms such as Android, iOS, web, Linux and macOS from the same environment. It would be nice to see a VM recognized in `flutter devices` and be able to run and debug from another host OS. I can imagine this might require something to be installed on the VM side to connect both ends together.

**Comment by stuartmorgan** [comment]
If this a feature request based on the assumption that cross-compilation is implemented for desktop? You've only discussed running, but it's not possible to build a Linux Flutter application on macOS either.

Or are you envisioning the build being done in the VM?

**Comment by dumazy** [comment]
I wasn't aware of the fact that it wouldn't be able to build on another host OS to be honest. I was simply envisioning developing for all platforms from one single machine. It's quite cumbersome to switch, definitely when you have a customized environment or IDE setup.

It would definitely be more challenging if it's not possible to build on a different OS. I few ideas come to mind, but I don't know which would be possible or considerable.
- Build on the target OS, but develop on another OS
- Build on the same OS, maybe within a container?

If one of those would be considerable (or any other solution) would it be possible to hot reload Dart code in debug mode, or would that also need to be compiled for a specific platform? Also, would it be possible to connect the debugger?
I think these features should still be available, else it's better to just develop on the targeted OS.

**Comment by Josua2012** [comment]
For example, in Delphi RAD Studio, development is always done from windows and by means of a program that can be installed on mac, linux, ... the development environment (windows) communicates automatically with mac simulators (iOS), etc.

**Comment by cloudwebrtc** [comment]
> I wasn't aware of the fact that it wouldn't be able to build on another host OS to be honest. I was simply envisioning developing for all platforms from one single machine. It's quite cumbersome to switch, definitely when you have a customized environment or IDE setup.

@dumazy I agree with your idea, and I thought of an idea, not using vm, but using a remote runner method,
For example, I develop a flutter app on mac, but on another linux or windows pc, install remote runner (a complete flutter development environment, a process for remote window capture), when I run flutter run -d remote_windows locally | remote_linux
A window will be created locally and connected to the flutter app window running on the remote windows through the network, and the debug port will be mapped to the local through the pipeline, and your mouse click on the window operation on the mac can also be passed to the remote runner.
I don't know if this idea is reasonable, but what I can confirm is that I can remotely display a specific window on another pc to my local mac.

**Comment by cloudwebrtc** [comment]
This has a problem, how to modify the flutter code on the mac and automatically synchronize to the directory where the windows/linux runner is located

**Comment by klausgreiner** [comment]
@cloudwebrtc at first we could have without hot reloading working, but theoretically you could open an tunnel with the target-machine to transfer necessary files. (I might be saying stupid stuff)
